src/core/encode_key_value.rs

```diff
@@ -154,13 +156,13 @@ impl fmt::Display for Data {
 struct KeyValueSerializer<'a> {
     key: String,
     separator: char,
-    output: &'a mut BTreeMap<String, Data>,
+    output: &'a mut BTreeMap<KeyString, Data>,
 }
 
 impl<'a> KeyValueSerializer<'a> {
-    fn new(key: String, separator: char, output: &'a mut BTreeMap<String, Data>) -> Self {
+    fn new(key: KeyString, separator: char, output: &'a mut BTreeMap<KeyString, Data>) -> Self {
         Self {
-            key,
+            key: key.into(),
```

Comment:
Is it possible to implicitly convert here?

Reply:
Oh, good catch. The field in the struct should have been a `KeyString` as seen below in `fn process` this results in a double conversion. Fixed in f0c79084b